core/includes/form.inc

```diff
@@ -2133,6 +2133,18 @@ function _form_builder_handle_input_element($form_id, &$element, &$form_state) {
       // be explicitly distinguished from missing input. (see below)
       $input_exists = NULL;
       $input = backdrop_array_get_nested_value($form_state['input'], $element['#parents'], $input_exists);
+      // Trim any leading/trailing whitespace from text input fields, unless
+      // their default '#trim' property has been overridden. To see which types
+      // of input elements are being trimmed by default, see
+      // system_element_info().
+      if (isset($element['#trim']) && $element['#trim'] != FALSE) {
```

Comment:
Might also be good to check that `$input` is actually a string here.

Reply:
A bit superfluous, since the `#trim` property is by default only added to form elements that have text input, but you can't be too sure I guess - lets not allow people to shoot themselves on the foot.

Reply:
...done with [`928b7b6` (#3845)](https://github.com/backdrop/backdrop/pull/3845/commits/928b7b60ed022d28f7477452b8b5c5d52ce3db1c)